17-lifecycle-lab/src/01-cps.jsx

```diff
@@ -1,18 +1,28 @@
-/**
- * Lab 01: Click per seconds
- *
- * Write a component that counts clicks per second.
- * Component should show a button and some text, and print
- * the click-per-seconds rate.
- */
-
-import React from 'react';
-
-export default createClass({
-  render() {
-    return (<div>
-      <button>Click Fast</button>
-      <p>CPS rate: {0}</p>
-    </div>);
-  }
-});
+import React from 'react';
+
+export default class ClicksCounter extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {counter: 0};
+    }
+
+    componentWillMount() {
+        this.timer = setInterval(() => {
+            this.setState({counter: 0});

```

Comment:
It's not a real cps. You clear count every second.
Let's say one second you click 4 times and then you clicked 0 time. Average CPS will be 2. In your case you will show 0. Which is incorrect.